src/Module.zig

```diff
@@ -2439,6 +2414,26 @@ pub const LazySrcLoc = union(enum) {
             },
         };
     }
+
+    pub fn fromZirInstData(inst_data: anytype) LazySrcLoc {
+        switch (@TypeOf(inst_data)) {
+            std.meta.FieldType(Zir.Inst.Data, .un_node),
+            std.meta.FieldType(Zir.Inst.Data, .pl_node),
+            std.meta.FieldType(Zir.Inst.Data, .int_type),
+            std.meta.FieldType(Zir.Inst.Data, .@"unreachable"),
+            std.meta.FieldType(Zir.Inst.Data, .inst_node),
+            => {
+                return LazySrcLoc.nodeOffset(inst_data.src_node);
+            },
+            std.meta.FieldType(Zir.Inst.Data, .un_tok),
+            std.meta.FieldType(Zir.Inst.Data, .pl_tok),
+            std.meta.FieldType(Zir.Inst.Data, .str_tok),
+            => {
+                return .{ .token_offset = inst_data.src_tok };
+            },
+            else => @compileError("Cannot create LazySrcLoc from " ++ @typeName(@TypeOf(inst_data))),
+        }
+    }
```

Comment:
I suspected you might be tempted to do it this way. Let's please keep it to having the same parameter type and same return type as before. So, you'll need to give names to those anonymous structs and make a function for each one. I recognize that this will have no impact on the machine code generated, however, I don't want to give up the simpler organization.